drivers/tapdisk-nbdserver.c

```diff
@@ -429,21 +430,63 @@ receive_newstyle_options(td_nbdserver_t *server, int new_fd, bool no_zeroes)
 			ERR("NBD_OPT_INFO: not implemented");
 			break;
 		case NBD_OPT_GO:
-		{   
+		{
 			uint16_t flags;
-			INFO("Processing NBD_OPT_EXPORT_NAME");
+			uint32_t exportnamelen;
+			uint16_t nrInfoReq;
+
+			INFO("Processing NBD_OPT_GO");
+
+			if (recv(new_fd, &exportnamelen, sizeof(exportnamelen), 0) == -1) {
```

Comment:
All these recv()s should ideally be a recv_exactly_n_bytes() function to avoid issues with short reads where the other side e.g. sends 1 byte at a time or doesn't send enough bytes.

They should also handle the case where the other side disconnects (receives 0 bytes).

Reply:
Will give that a try in the next set of test runs.

Reply:
There's `readit()` and `sendit()` in drivers/block-nbd.c that could be converted to utility functions. Though they need to be fixed... readit() is missing a break for the error case and both of them check `errno` when the send()/recv() returns 0 which does not set errno. That means that if errno happens to be EAGAIN from some previous call it will have an endless loop.